app/views/layouts/application.html.erb

```diff
@@ -99,6 +99,14 @@
       end
     end %>
 
+    <% unless current_page?(notifications_path) %>
+      <% if current_user&.active_unread_notifications.present? %>
+        <%= render "notifications/notification_banner", notification_count: current_user.active_unread_notifications.count, notification: current_user.newest_active_unread_notification %>
+      <% elsif current_user.blank? && Notification.active_unauthenticated_notifications.present? %>
+        <%= render "notifications/notification_banner", notification_count: Notification.active_unauthenticated_notifications.count, notification: Notification.newest_active_unauthenticated_notification %>
+      <% end %>
+    <% end %>
+
```

Comment:
Would it make sense to move this to notification_banner and just have the render here?

Reply:
I think it would be a bit more confusing to do that, as the notification_banner only renders when there are notifications so I'd like to keep at least a condition representing that here. I'll look at assigning the local variables intelligently so I can just have one render statement which would be clearer